hoomd/mpcd/DryingDropletStreamingMethod.h

```diff
@@ -0,0 +1,311 @@
+    };
+
+/*!
+ * \param timestep Current time to stream
+ */
+void DryingDropletStreamingMethod::stream(unsigned int timestep)
+    {
+    // use peekStream since shouldStream will be called by parent class 
+    if(!peekStream(timestep)) return;
+    
+    // compute final Radius and Velocity of surface
+    const Scalar start_R = m_R->getValue(timestep);
+    const Scalar end_R = m_R->getValue(timestep + m_period);
+    const Scalar V = (end_R - start_R)/(m_mpcd_dt);
+    // checks if V <= 0, since size of droplet must decrease
+    if (V > 0)
+        {
+        throw std::runtime_error("Droplet radius must decrease.");
+        }
+
+    /*
+     * If the initial geometry was not set. Set the geometry and validate the geometry 
+     * Because the interface is shrinking, it is sufficient to validate only the first time the geometry
+     * is set.
+     */
+    if (!m_geom)
+        {
+        m_geom = std::make_shared<mpcd::detail::SphereGeometry>(start_R, V , m_bc ); 
+        validate();
+        m_validate_geom = false;
+        }
+
+    /*
+     *Update the geometry radius to the size at the end of the streaming step.
+     * This needs to be done every time.
+     */
+    m_geom = std::make_shared<mpcd::detail::SphereGeometry>(end_R, V, m_bc );
+    
+    //stream according to base class rules
+    ConfinedStreamingMethod<mpcd::detail::SphereGeometry>::stream(timestep);
+
+    //calculating number of particles to evaporate(such that solvent density remain constant)
+    const unsigned int N_remove = round((4.*M_PI/3.)*end_R*end_R*end_R*m_density);
+    const unsigned int N_evap = m_mpcd_pdata->getNGlobal() - N_remove;
+
+    // get the compact array of indexes of bounced particles and total N_bounced
+    unsigned int N_bounced = calculateNumbounced();
+
+    // reduce / scan the number of particles that were bounced on all ranks
+    unsigned int N_bounced_total = N_bounced;
+    unsigned int N_before = 0;
+    #ifdef ENABLE_MPI
+    if (m_exec_conf->getNRanks() > 1)
+        {
+        MPI_Allreduce(&N_bounced, &N_bounced_total, 1, MPI_UNSIGNED, MPI_SUM, m_exec_conf->getMPICommunicator());
+        MPI_Exscan(&N_bounced, &N_before, 1, MPI_UNSIGNED, MPI_SUM, m_exec_conf->getMPICommunicator());
+        }
+    #endif // ENABLE_MPI
+
+    /* if N_bounced_total <= N_evap, pick all the particles to delete
+     * else pick N_evap particles out of N_bounced_total
+     */
+    if (N_bounced_total <= N_evap)
+        {
+        //pick all bounced particles and remove all of them
+        m_picks.resize(N_bounced);
+        ArrayHandle<unsigned int> h_picks(m_picks, access_location::host, access_mode::overwrite);
+        std::iota(h_picks.data, h_picks.data + N_bounced, 0);
+        m_Npick = N_bounced;
+        //calculating density if it will get changed alot, print a warning
+        Scalar currentdensity = (m_mpcd_pdata->getNGlobal() - N_bounced_total)/((4.*M_PI/3.)*end_R*end_R*end_R);
```

Comment:
Maybe you should compute the end volume `V_end` above and save in a variable so it isn't floating around like a magic formula here?
```suggestion
        Scalar currentdensity = (m_mpcd_pdata->getNGlobal() - N_bounced_total)/V_end;
```